### ISSUE TITLE
Question: can number of replicas be used for batch size != 1

### ISSUE THREAD

**Issue by dylanabramson33** [torchmd/torchmd#19]
I am working with autoencoders on MD structures and I am attempting to use the energy of generated structures as an additional loss term in addition to the reconstruction error. For handling multiple structures per batch, I just set the number of replicas to the number of samples per batch. Does this seem like a reasonable approach and are there any pitfalls I might not be considering? Calculating the potential energy returns a scalar, so I'm not sure if autograd is working under the hood the way I think it is. I am getting pretty good decoded structures but do have some slight issues.

TorchMD system initialization:
```python 
testdir = "./DATA"
mol = Molecule(os.path.join(testdir, "adp2.prmtop"))  # Reading the system topology

precision = torch.float
device = "cuda:0"
system = System(mol.numAtoms, nreplicas=batch_size, precision=precision, device=device)
ff = ForceField.create(mol, os.path.join(testdir, "adp2.prmtop"))
parameters = Parameters(ff, mol, precision=precision, device=device)
```

Here is the interior of my training loop:
```python
reconstruction = autoencoder(data)
Epot = forces.compute(reconstruction, system.box, system.forces, returnDetails=False,explicit_forces=False)
dist = torch.cdist(reconstruction,reconstruction)
true_dist = torch.cdist(data,data)
new_dist = dist - true_dist
fro_norm = torch.norm(new_dist, p='fro', dim=(1,2))
loss = (1/temperature)*Epot+1/4*torch.sum(fro_norm)
index += 1
loss.backward()
optimizer_ae.step()
```

**Comment by stefdoerr** [comment]
Hi, sorry for the delay in this. I checked the code and indeed if you use `explicit_forces=False` it returns a scalar which is the sum of energies of all your systems. Running backwards on it indeed should return the correct forces so there is no issue there.
I changed the code now to return the energy for each system so that it's consistent with `explicit_forces=True`
But as I mentioned, it's probably not work running multiple replicas. I think we added this so that we could easier run multiple replicas on a single device.
